chain/manager.go

```diff
@@ -1001,44 +1007,56 @@ func (m *Manager) V2TransactionSet(basis types.ChainIndex, txn types.V2Transacti
 	return m.tipState.Index, append(parents, txn), nil
 }
 
-func (m *Manager) checkDupTxnSet(txns []types.Transaction, v2txns []types.V2Transaction) (types.Hash256, bool) {
+func (m *Manager) checkTxnSet(txns []types.Transaction, v2txns []types.V2Transaction) (bool, error) {
 	allInPool := true
-	checkPool := func(txid types.TransactionID) {
+	checkPool := func(txid types.TransactionID) types.TransactionID {
 		if allInPool {
 			if _, ok := m.txpool.indices[txid]; !ok {
 				allInPool = false
 			}
 		}
+		return txid
 	}
 	h := types.NewHasher()
-	for i, txn := range txns {
-		txid := txn.ID()
-		checkPool(txid)
-		h.E.WriteUint64(uint64(i))
-		txid.EncodeTo(h.E)
+	for _, txn := range txns {
+		checkPool(txn.ID()).EncodeTo(h.E)
 	}
-	for i, txn := range v2txns {
-		txid := txn.ID()
-		checkPool(txid)
-		h.E.WriteUint64(uint64(i))
-		txid.EncodeTo(h.E)
+	for _, txn := range v2txns {
+		checkPool(txn.ID()).EncodeTo(h.E)
 	}
 	setID := h.Sum()
```

Comment:
Would there be any benefit in making the order of the transactions within the set deterministic to make sure the order doesn't influence the set id?

Reply:
pretty small, I think; most of the time, transactions can't be reordered anyway due to dependencies.

(In fact, I wonder if we should *reject* transaction sets that *aren't* internally dependent...hmm. Probably not worth it.)